lib/src/ovis_util/util.h

```diff
@@ -268,13 +273,20 @@ int f_mkdir_p(const char *path, __mode_t mode);
  *
  * The supported syntax for the <name> is [[:alnum:]_]+. That is, one
  * or more alpha-numeric or '_' characters. Environment variables that
- * are missing are replaced with "".
+ * are missing are replaced with "". Similarly, ~{key} is replaced with
+ * the value of its first occurence in the avl, or the empty string if it is
+ * undefined.
+ *
+ * All tilde substitions are processed before any environment
+ * variable substitutions. If recursion among 2 or more ~{} substitutions
+ * is detected, the values for the involved keys are undefined.
  *
  * The function returns memory that was allocated with malloc()
  *
  * \param str The input string.
- * \retval NULL There was insufficient memory to allocate the output string.
- * \retval Ptr to a string with the environment variable values replaced.
+ * \retval NULL There was insufficient memory to allocate the output string
+ * or recursion was detected; errno will be ENOMEM or EBADSLT respectively.
+ * \retval Ptr to a string with the environment variable and ~{} values replaced.
  */
 char *str_repl_env_vars(const char *str);
```

Comment:
i'm not picky about the name. But the current name has been part of the public API for a while so the rename could break code we don't know about.  The extension of the semantics is unlikely to affect anyone since ~{xxx} is not a common pattern of naming things likely to be on the right side of attr=value.

Reply:
@nichamon, @baallan should I accept this pull request or do we still have design issues to resolve?